src/commands/snippets/snippets.ts

```diff
@@ -0,0 +1,118 @@
+
+interface Snippet {
+  prefix: string
+  body: string
+  description: string[]
+}
+
+const sasRegExp = /\.sas$/ // to check if file has .sas extension
+const briefRegExp = /^\s\s@brief\s/ // to get lines that has @brief keyword
+const paramRegExp = /^\s\s@param\s/ // to get lines that has @param keyword
+
+/**
+ * Generates VS Code snippets from the Doxygen headers in the SAS Macros.
+ * @param macroFolders an array of file paths of SAS Macros.
+ * @param outDirectory a folder path where generated snippets should be put into. If ends with '<file name>.json' then provided file name will be used, otherwise default file name ('sasjs-macro-snippets.json') will be used.
+ * @returns a promise that resolves into a file path with generated VS Code snippets.
+ */
+export async function generateSnippets(
+  macroFolders: string[],
+  outDirectory?: string
+) {
+  // an object that represents generated snippets
+  const snippets: { [key: string]: Snippet } = {}
+
+  // return an error if no macro folders has been provided
+  if (!macroFolders.length) {
+    return Promise.reject(
+      `"macroFolders" array was not found in sasjs/sasjsconfig.json.`
+    )
+  }
+
+  // generate snippets from all .sas file in macro folders
+  await asyncForEach(macroFolders, async (folder) => {
+    // get .sas files excluding test('*.test.sas') files
+    const sasFiles = (await listSasFilesInFolder(folder, true))
+      .map((file) => path.join(folder, file))
+      .filter((file) => !isTestFile(file))
+
+    await asyncForEach(sasFiles, async (file) => {
+      // get macro name
+      const macro: string = file.split(path.sep).pop().replace(sasRegExp, '')
+
+      // put generated snippet into snippets object
+      snippets[macro] = await createMacro(file)
+    })
+  })
+
+  // return an error if no snippets has been generated
+  if (!Object.keys(snippets).length) {
+    return Promise.reject('No VS Code snippets has been found.')
+  }
+
+  const defaultOutputFileName = 'sasjs-macro-snippets.json'
+  const { buildDestinationResultsFolder } = process.sasjsConstants
+
+  // if outDirectory is provided, use it depending if it has file name or a folder only. If outDirectory is not provided, default file name and build result folder should be used.
+  const snippetsFilePath = path.join(
+    outDirectory ? process.projectDir : buildDestinationResultsFolder,
+    outDirectory
+      ? /\.json$/.test(outDirectory)
+        ? outDirectory
+        : path.join(outDirectory, defaultOutputFileName)
+      : defaultOutputFileName
+  )
+
+  // create file with generated VS Code snippets
+  await createFile(snippetsFilePath, JSON.stringify(snippets, null, 2))
+
+  // return file path with generated VS Code snippets
+  return Promise.resolve(snippetsFilePath)
+}
+
+/**
+ * Creates a VS Code snippet from SAS macro.
+ * @param file file path of SAS macro.
+ * @returns promise that resolves with VS Code snippet.
+ */
+const createMacro = async (file: string): Promise<Snippet> => {
```

Comment:
I think the function name should be updated. Add something about the snippet.

Reply:
makes sense, fixed